airflow/models/__init__.py

```diff
@@ -688,6 +632,57 @@ def __init__(self, task, execution_date, state=None):
         # Not persisted to the database so only valid for the current process
         self.raw = False
 
+    @staticmethod
```

Comment:
what is the main purpose of moving this function to a static method here which bascially serve the same purpose?

Reply:
mainly because it never uses `self` methods. Now the method can be used with a instance of TaskInstance itself. Makes more sense because this method does multiple TaskInstances and not only the instance where you calling the method on